rules/php74/tests/Rector/Property/TypedPropertyRector/FixtureDoctrine/doctrine_one_to_one_class_keyword.php.inc

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rector\Php74\Tests\Rector\Property\TypedPropertyRector\Fixture;
+
+use Doctrine\ORM\Mapping as ORM;
+
+class DoctrineOneToManyClassKeywordClass
+{
+    /**
+     * @ORM\OneToOne(targetEntity=Rector\Php74\Tests\Rector\Property\TypedPropertyRector\Source\SomeParent::class)
+     * @ORM\JoinColumn(nullable=false)
+     */
+    private $package;
+}
+?>
+-----
+<?php
+
+namespace Rector\Php74\Tests\Rector\Property\TypedPropertyRector\Fixture;
+
+use Doctrine\ORM\Mapping as ORM;
+
+class DoctrineOneToManyClassKeywordClass
+{
+    /**
+     * @ORM\OneToOne(targetEntity=Rector\Php74\Tests\Rector\Property\TypedPropertyRector\Source\SomeParent::class)
+     * @ORM\JoinColumn(nullable=false)
+     */
+    private \Rector\Php74\Tests\Rector\Property\TypedPropertyRector\Source\SomeParent $package;
```

Comment:
```suggestion
    /**
     * @ORM\OneToOne(targetEntity=SomeParent::class)
     * @ORM\JoinColumn(nullable=false)
     */
    private \Rector\Php74\Tests\Rector\Property\TypedPropertyRector\Source\SomeParent $package;
```

There should be 1 case for imported class too

Reply:
Are you sure? :) I checked some tests and all of them use FQCN... When I use the imported class name it always adds `\` before the class name to make it kinda FQCN and do not resolve it correctly :/ and the bug is actually here
https://github.com/rectorphp/rector/pull/3770/files#diff-5641698a38562059eecf39ef68b4b9efR57

`$node->getAttribute(AttributeKey::NAMESPACE_NAME)` returns the namespace of the current test fixture class and not the namespace of the imported class, thus `class_exists` will always return `false`

```diff
-Rector\Php74\Tests\Rector\Property\TypedPropertyRector\Fixture
+Rector\Php74\Tests\Rector\Property\TypedPropertyRector\Source
```

Reply:
Oh, it's broken in the same as in input. I haven't noticed that.

If it works, then it's ok :+1: